doc/source/data/user-guide.rst

```diff
@@ -15,7 +15,7 @@ show you how achieve several tasks.
     loading-data
     transforming-data
     consuming-data
-    batch_inference
     working-with-tensors
     data-internals
     performance-tips
+    batch_inference
```

Comment:
why move one of the top 3 CUJs on a company level to the very bottom?

Reply:
Per offline discussion, we're structuring the user guides from basic -> intermediate -> e2e.

This guide will still be linked from the necessary places.

![image](https://github.com/ray-project/ray/assets/8068268/29689ea8-6d43-42e2-a8e9-1542e177970e)

Reply:
cc @bveeramani

Reply:
I think we were only going to add a prefix for the batch inference guide.

The "Working with ..." guides aren't really conceptual anyway.

Reply:
that sgtm!

Reply:
sgtm, 2!